importlib_metadata/__init__.py

```diff
@@ -831,7 +831,13 @@ def __init__(self, path: FastPath):
         self.infos = FreezableDefaultDict(list)
         self.eggs = FreezableDefaultDict(list)
 
-        for child in path.children():
+        for child in sorted(path.children()):
```

Comment:
The Lookup is part of FastPath, which is highly optimized. Throwing a sort in here could noticeably affect performance. What's the purpose of sorting? What bug does it fix?